mediagrains_py36/numpy/__init__.py

```diff
@@ -0,0 +1,220 @@
+            return np.dtype(np.int32)
+    elif fmt in [CogFrameFormat.UYVY,
+                 CogFrameFormat.YUYV,
+                 CogFrameFormat.RGB,
+                 CogFrameFormat.RGBx,
+                 CogFrameFormat.RGBA,
+                 CogFrameFormat.BGRx,
+                 CogFrameFormat.BGRx,
+                 CogFrameFormat.ARGB,
+                 CogFrameFormat.xRGB,
+                 CogFrameFormat.ABGR,
+                 CogFrameFormat.xBGR]:
+        return np.dtype(np.uint8)
+    elif fmt == CogFrameFormat.v216:
+        return np.dtype(np.int16)
+    elif fmt == CogFrameFormat.v210:
+        return np.dtype(np.int32)
+
+    raise NotImplementedError("Cog Frame Format not amongst those supported for numpy array interpretation")
+
+
+class ComponentDataList(list):
+    class ComponentOrder (Enum):
+        YUV = auto()
+        RGB = auto()
+        BGR = auto()
+        X   = auto()
+
+    def __init__(self, data: list, arrangement: ComponentOrder=ComponentOrder.X):
+        super().__init__(data)
+        if arrangement == ComponentDataList.ComponentOrder.YUV:
+            self.Y = self[0]
+            self.U = self[1]
+            self.V = self[2]
+        elif arrangement == ComponentDataList.ComponentOrder.RGB:
+            self.R = self[0]
+            self.G = self[1]
+            self.B = self[2]
+        elif arrangement == ComponentDataList.ComponentOrder.BGR:
+            self.B = self[0]
+            self.G = self[1]
+            self.R = self[2]
+
+
+def _component_arrangement_from_format(fmt: CogFrameFormat):
+    if not COG_FRAME_IS_PACKED(fmt) and not COG_FRAME_IS_COMPRESSED(fmt):
+        if COG_FRAME_IS_RGB(fmt):
+            return ComponentDataList.ComponentOrder.RGB
+        else:
+            return ComponentDataList.ComponentOrder.YUV
+    elif fmt in [CogFrameFormat.UYVY, CogFrameFormat.YUYV, CogFrameFormat.v216]:
+        return ComponentDataList.ComponentOrder.YUV
+    elif fmt in [CogFrameFormat.RGB, CogFrameFormat.RGBA, CogFrameFormat.RGBx, CogFrameFormat.ARGB, CogFrameFormat.xRGB]:
+        return ComponentDataList.ComponentOrder.RGB
+    elif fmt in [CogFrameFormat.BGRA, CogFrameFormat.BGRx]:
+        return ComponentDataList.ComponentOrder.BGR
+    else:
+        return ComponentDataList.ComponentOrder.X
+
+
+def _component_arrays_for_data_and_type(data: np.ndarray, fmt: CogFrameFormat, components: bytesgrain.VIDEOGRAIN.COMPONENT_LIST):
+    if not COG_FRAME_IS_PACKED(fmt) and not COG_FRAME_IS_COMPRESSED(fmt):
+        arrays = []
+        for component in components:
+            component_data = data[component.offset//data.itemsize:(component.offset + component.length)//data.itemsize]
+            component_data = as_strided(component_data, shape=(component.height, component.width), strides=(component.stride, component_data.itemsize))
+            arrays.append(component_data.transpose())
+        return arrays
+    elif fmt in [CogFrameFormat.UYVY, CogFrameFormat.v216]:
```

Comment:
The block that follows this feels a bit messy, could it be re-factored into a method that accepts a block of data, the width and itemsize for it?